File: src/main/java/com/leet/medium/BrickWall.go
package main

func leastBricks(wall [][]int) int {
  width := 0
  count := map[int]int{}
  for _, bricks := range wall {
    pos := 0
    for _, brick := range bricks {
      pos += brick
      count[pos]++
    }
    width = pos
  }
  count[width]=0
  ans := 0
  for _, value := range count {
    if value > ans {
      ans = value
    }
  }
  return len(wall) - ans
}

File: src/main/java/com/leet/medium/BrickWall.java
package com.leet.medium;

import java.util.HashMap;
import java.util.List;

public class BrickWall {
  public int leastBricks(List<List<Integer>> wall) {
    var edges = new HashMap<Integer, Integer>();
    int end = 0;
    for (var bricks : wall) {
      int pos = 0;
      for (var brickWidth : bricks) {
        pos += brickWidth;
        edges.merge(pos, 1, Integer::sum);
      }
      end = pos;
    }

    int ans = 0;
    for (var entry : edges.entrySet()) {
      if (entry.getKey() == end) continue;
      ans = Math.max(ans, entry.getValue());
    }
    return wall.size() - ans;
  }
}
